addon/-private/state-machine/-base.ts

```diff
@@ -56,123 +56,61 @@ export default abstract class StateMachine extends EmberObject {
   }
 
   addStep(this: StateMachine, name: string) {
-    if (this.stepsToAdd.includes(name)) {
+    if (this.stepTransitions.includes(name)) {
       return;
     }
 
-    this.stepsToAdd.push(name);
+    this.stepTransitions.pushObject(name);
+
     return bind(
       this,
       scheduleOnce,
-      'afterRender',
+      'actions',
       this,
-      this.flushAdditionQueue
+      this.updateFirstCurrentAndLastSteps
     )();
   }
 
   removeStep(this: StateMachine, name: string) {
-    if (this.stepsToRemove.includes(name)) {
+    if (!this.stepTransitions.includes(name)) {
       return;
     }
 
-    this.stepsToRemove.push(name);
-    return bind(this, scheduleOnce, 'render', this, this.flushRemoveQueue)();
-  }
+    this.stepTransitions.removeObject(name);
 
-  flushAdditionQueue(this: StateMachine) {
-    let { firstStep, currentStep, stepsToAdd, stepTransitions } = getProperties(
+    return bind(
```

Comment:
Is this thing still necessary? It really doesn't feel right. Why is `removeStep` returning a function? It's never called.

Reply:
It gets called right away. I did try to remove the bind but I still get the same ember runloop error. It might be able to be moved, but it would basically just moving the ember runloop error to the next time a unit test is written.